spec/classes/collectd_plugin_mongodb_spec.rb

```diff
@@ -0,0 +1,206 @@
+    it { should contain_file('mongodb.load').with_content(default_fixture) }
+  end
+
+  describe 'with ensure parameter' do
+    %w(present absent).each do |value|
+      context "set to a valid value of #{value}" do
+        let :params do
+          default_params.merge(:ensure => "#{value}",)
+        end
+
+        it do
+          should contain_file('mongodb.load').with(
+            'ensure' => value,
+          )
+        end
+      end
+    end
+
+    context 'set to an invalid value' do
+      let :params do
+        default_params.merge(:ensure => 'invalid',)
+      end
+
+      it 'should fail' do
+        expect do
+          should contain_class('collectd::plugin::mongodb')
+        end.to raise_error(Puppet::Error, /collectd::plugin::mongodb::ensure is <invalid> and must be either 'present' or 'absent'\./)
+      end
+    end
+  end
+
+  describe 'with interval parameter' do
+    ['10.0', '3600'].each do |value|
+      context "set to a valid numeric of #{value}" do
+        let(:params) do
+          default_params.merge(:interval => "#{value}",)
+        end
+
+        it { should contain_file('mongodb.load').with_content(/\s*Interval\s+#{Regexp.escape(value)}/) }
+      end
+    end
+
+    context 'set to an invalid value' do
+      let :params do
+        default_params.merge(:interval => 'invalid',)
+      end
+
+      it 'should fail' do
+        expect do
+          should contain_class('collectd::plugin::mongodb')
+        end.to raise_error(Puppet::Error, /Expected first argument to be a Numeric or Array, got String/)
+      end
+    end
+  end
+
+  describe 'with db_host parameter' do
+    context 'set to a valid IP' do
+      let :params do
+        default_params.merge(:db_host => '127.0.0.1',)
+      end
+
+      hostdb_fixture = File.read(fixtures('plugins/mongodb.conf.hostdb'))
+      it { should contain_file('mongodb.load').with_content(hostdb_fixture) }
+    end
+
+    %w('127001', nil).each do |value|
+      context 'set to and invalid IP' do
+        let :params do
+          default_params.merge(:db_host => "#{value}",)
+        end
+
+        it 'should fail' do
+          expect do
+            should contain_class('collectd::plugin::mongodb')
+          end.to raise_error(Puppet::Error, /must be a valid IP address/)
+        end
+      end
+    end
+  end
+
+  describe 'with db_user parameter' do
+    context 'set to a valid user name' do
+      let :params do
+        default_params.merge(:db_user => 'test_user',)
+      end
+
+      confdb_fixture = File.read(fixtures('plugins/mongodb.conf.db_user'))
+      it { should contain_file('mongodb.load').with_content(confdb_fixture) }
+    end
+
+    context 'undefined' do
+      let :params do
+        { :db_pass => 'password',
+        }
+      end
+
+      it 'should fail' do
+        expect do
+          should contain_class('collectd::plugin::mongodb')
+        end.to raise_error(Puppet::Error, /is <undef> and must be a mongodb username/)
```

Comment:
Is this from the semantics of the mongodb plugin? With `mongodb` itself, it's (unfortunately) quite common to use no authentication, isn't it?